client/src/plugins/usage-statistics/event-handlers/BaseEventHandler.js

```diff
@@ -29,10 +29,10 @@ export default class BaseEventHandler {
     return this._isEnabled;
   }
 
-  enable() {
+  async enable() {
```

Comment:
Making this async is most likely inevitable, but the async should be followed through to the source otherwise we're going to get some potential mystery behaviors in the future.

In UsageStatistics.js, we should be awaiting eventHandler.enable() in the loop, otherwise we're just firing off promises we never check on again, which is unsafe. 

This means enable needs to be async. But you'll see that every call to it is async aswell which means we won't need to dig any deeper with this.

Reply:
Note that my knowledge of asynchronous programming stems from C# but I believe the principles should be the same here 😄

Reply:
You mean here?
https://github.com/camunda/camunda-modeler/blob/d44b3a0fa258caa83093b17b058db83ad753e05e/client/src/plugins/usage-statistics/UsageStatistics.js#L77

I don't see a reason why we need to `await eventHandler.enable()`, since we don't do anything with it, or anything that requires that to have finished. Can you explain?

Reply:
We had a quick sync about this, just writing for visibility.

Basically we'll take a fire and forget approach, we don't need to await this.onAfterEnable() and hence won't need to make onEnable async, which simplifies everything. I'm going back on my initial statement that it's unsafe to do fire and forget because it does seem that these errors still get caught at some point. Note that in the future if we'll want more consistency in how these plugins load in, we'll need to do async all the way.

Reply:
Actually, I just had a look. The reasoning for having `async enable()` and `await this.onAfterEnable()` is because of the tests. So we could do `await enable()`, and check that it was sent

Reply:
We will leave as is